### PR TITLE
avoid tip closing on before_close callback

### PR DESCRIPTION
**Pull request by elboletaire**
Simply add a return false on before_close method:

 $.gritter.add({
    title: "uploading..",
    text: fileObj.name,
    sticky: true,
    class_name: id,
    before_close: function(e, manual) {
        if ( manual ) {
            if ( confirm("Are you sure you want to close the tip??") ) {
                // do something
            } else return false; // avoid the tip closing
        }
    }
});

### CODE REVIEW

**Comment by jboesch** [comment]
A few things:
1. Can you add your example in the comments to index.html?
2. Remove spaces between brackets.

File: js/jquery.gritter.js [hunk 1]
```diff
@@ -202,7 +202,9 @@
 				fade_out_speed = params.speed || this.fade_out_speed,
                 manual_close = unbind_events;
 			
-			this['_before_close_' + unique_id](e, manual_close);
+			if ( this['_before_close_' + unique_id](e, manual_close) === false ) {
```
Comment (jboesch) [inline js/jquery.gritter.js]: can you remove spacing between brackets... to be
if (this['_before_close_' + unique_id](e, manual_close) === false){

**Comment by elboletaire** [comment]
I'm not sure if I'm putting the comment in the right place...

I've added these lines:

// if this function return false the notification won't be closed. e.g...
//if (manual_close) {
      // return confirm("Are you sure?");
//}

Under the first before_close callback (line 155 aprox.), and this...

// if this function return false the notification won't be closed.

unser the second before_close callback (line 185 aprox.)

I've done a commit to master and as I can see it has been merged with my 
latest pull request. Tell me if I done it wrong.

https://github.com/jboesch/Gritter/pull/6

Al 08/05/2011 21:18, En/na jboesch ha escrit:

> A few things:
> 1. Can you add your example in the comments to index.html?
> 2. Remove spaces between brackets.

**Comment by jboesch** [comment]
Will add this later.

**Pull request closed without merging**